### ISSUE TITLE
Dead lock on shutdown

### ISSUE THREAD

**Issue by StefanBertels** [louthy/echo-process#69]
I found a dead lock bug but currently have no clear idea how to fix this properly.

Here is a reproducing test:

```
public void SystemShutdownDuringActorSetup()
            {
                Echo.ProcessConfig.initialise();
                {
                    var actor = spawn("test", () =>
                    {
                        Thread.Sleep(1 * seconds);
                        throw new Exception();
                        return Disposable.Empty;
                    });
                    Thread.Sleep(0.5 * seconds);
                    shutdownAll();
                }
                // will not arrive here
            }
```

The problem is this line:

https://github.com/louthy/echo-process/blob/master/Echo.Process/ActorSys/Actor.cs#L1016

`Parent.Actor.UnlinkChild` will call `lock(sync)` of the parent of the test actor. Test actor itself is already locked. This means: child is locked before parent. Situation is a result of the exception thrown in setup function, which calls RunStrategy which might call shutdown (Directive.Stop).

The parallel `shutdownAll` will lock actors in this order: First the parent is locked and then the children will be locked (hierarchical shutdown).

**Comment by StanJav** [comment]
Paul this is exactly what I found by calling 'kill' and 'spawn' for a child process. I will put together an unit test for that.

**Comment by louthy** [comment]
Fixed in `2.0.69` - thanks @StanJav
